plugins/sinks/gcs/client.go

```diff
@@ -0,0 +1,49 @@
+package gcs
+
+import (
+	"context"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+	"google.golang.org/api/option"
+)
+
+type GCSClient interface {
+	WriteData([]byte) error
+	Close() error
+}
+
+type gcsClient struct {
+	client *storage.Client
+	writer *storage.Writer
+}
+
+func newGCSClient(ctx context.Context, serviceAccountJSON []byte, bucketname string, filepath string) (GCSClient, error) {
+	client, err := storage.NewClient(ctx, option.WithCredentialsJSON(serviceAccountJSON))
+	if err != nil {
+		return nil, errors.Wrap(err, "error in creating client")
+	}
+
+	writer := client.Bucket(bucketname).Object(filepath).NewWriter(ctx)
+
+	return &gcsClient{
+		client: client,
```

Comment:
I don't see client getting used anywhere, should we get rid of it?

Reply:
yes, its's only used to create the writer, will update the struct